src/sentry/api/serializers/models/integration.py

```diff
@@ -89,22 +92,32 @@ def serialize(self, obj, attrs, user, include_config=True):
             include_config=include_config,
         )
 
-        # TODO: skip adding configData if include_config is False
-        # we need to wait until the Slack migration is complete first
-        # because we have a dependency on configData in the integration directory
+        dynamic_display_information = None
+
         try:
             installation = obj.integration.get_installation(obj.organization_id)
         except NotImplementedError:
             # slack doesn't have an installation implementation
-            config_data = obj.config
-            dynamic_display_information = None
+            config_data = obj.config if include_config else None
         else:
-            # just doing this to avoid querying for an object we already have
-            installation._org_integration = obj
-            config_data = installation.get_config_data()
-            dynamic_display_information = installation.get_dynamic_display_information()
+            try:
+                # just doing this to avoid querying for an object we already have
+                installation._org_integration = obj
+                config_data = installation.get_config_data() if include_config else None
+                dynamic_display_information = installation.get_dynamic_display_information()
+            except ApiError as e:
+                # If there is an ApiError from our 3rd party integration providers, assume there is an problem with the configuration and set it to disabled.
+                integration.update({"status": "disabled"})
+                name = "sentry.serializers.model.organizationintegration"
+                log_info = {
+                    "error": e,
```

Comment:
@NisanthanNanthakumar I think we should call `str(e)` here to avoid passing in an object here.